File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let listLength = 0;
    let find = 0;
    let start1 = head;
    while (start1){
        listLength++;
        start1 = start1.next;
    }
    
    if (listLength/2 === Math.trunc(listLength/2)){
        find = (listLength/2)+1;
    } else {
        find = (listLength/2) +0.5;
    }
    
    let start2 = head;
    for (let i=0; i<find-1; i++){
        start2 = start2.next;
    }
    
    return start2;
};

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        count = 0
        start1 = head
        
        find = 0
        
        while start1:
            count += 1
            start1 = start1.next
            
        print(f"the linked list has {count} entries")
        if (count/2) == int(count/2):
            print(count)
            find = (count/2) +1
            print('int');
        else:
            find = (count/2) + 0.5
            print(count)
            print('not int')
        
        
        start2 = head
        print(find)
        for i in range(int(find)-1):
            start2 = start2.next
        
        return start2